coreax/kernel.py

```diff
@@ -915,6 +901,10 @@ def compute_elementwise(
         :param y: Vector :math:`\mathbf{y} \in \mathbb{R}^d`
         :return: Kernel evaluated at (``x``, ``y``)
         """
+        if not isinstance(self.base_kernel, Kernel):
```

Comment:
As far as I'm aware, the preferred approach on Coreax at the moment is not to rely on `isinstance` checks, but rather to rely on `hasattr` checks (i.e. relying on duck typing rather than explicit typing). I'm happy for this change to go through, but you should probably get a second opinion!

Reply:
What do you think @tp832944 ?

Reply:
The agreed method (see https://github.com/gchq/coreax/issues/420#issuecomment-1961033259) is to just let it crash when safe to do so. Otherwise, use `try ... except`. If the error message from letting it crash is unhelpful, use `try ... except` then reraise a useful error. In this case, the attribute error raised by the next line (908) is fairly informative, so just delete 904-907.

Reply:
On the topic of helpful vs unhelpful error messages, I seem to remember that trying to JIT an object missing `tree_flatten` or `tree_unflatten` gave a really unhelpful error message. However, if we're moving to `equinox` this shouldn't be an issue.

Reply:
I'll remove the validation change from this pull and we can have any other discussions regarding validation in #583